File: exercism/clojure/grade-school/src/grade_school.clj
(ns grade-school)

(defn grade [school grade]
  (get school grade []))

(defn add [school name grade]
  (update school grade concat (vector name)))

(defn sorted [school]
  (let [sorted-school (into (sorted-map) (sort-by key school))] (reduce-kv #(assoc %1 %2 (sort (vec %3))) (empty sorted-school) sorted-school)))

File: exercism/python/grade-school/grade_school.py
class School:
    def __init__(self):
        self.students_by_grade = {}

    def add_student(self, name, grade):
        if grade not in self.students_by_grade:
            self.students_by_grade[grade] = []
        
        self.students_by_grade[grade].append(name)

    def roster(self):
        return [student for _, students in sorted(self.students_by_grade.items()) for student in sorted(students)]

    def grade(self, grade_number):
        return sorted(self.students_by_grade[grade_number]) if grade_number in self.students_by_grade else []
